Fix unencountered bug if multiple Primaries in same process

awwad opened the pull request:
If multiple Primaries are initialized in the same process and
both are initialized without specifying a list of their
Secondaries, then they will both point to the same list of
Secondaries, updating together, rather than having distinct
lists. This is a Python quirk related to the use of mutables
as default arguments in a function. Mutables should not be
default values because they become common across all calls
made to that function. (It is unlikely that this would have
occurred for anyone, unlikely that any implementer would
inherit this issue in another language, and unlikely to run
multiple Primaries in the same process regardless; however,
it is best to avoid this anti-pattern in Python.)

awwad committed:
Fix unencountered bug if multiple Primaries and optional arg

If multiple Primaries are initialized in the same process and
both are initialized without specifying a list of their
Secondaries, then they will both point to the same list of
Secondaries, updating together, rather than having distinct
lists. This is a Python quirk related to the use of mutables
as default arguments in a function. Mutables should not be
default values because they become common across all calls
made to that function. (It is unlikely that this would have
occurred for anyone, unlikely that any implementer would
inherit this issue in another language, and unlikely to run
multiple Primaries in the same process regardless; however,
it is best to avoid this anti-pattern in Python.

Signed-off-by: Sebastien Awwad <sebastien.awwad@gmail.com>

awwad commented:
(Thanks go to Lukas for bringing this Python anti-pattern to my attention in a different context.)

vladimir-v-diaz commented on uptane/clients/primary.py:
```diff
@@ -243,7 +243,8 @@ def __init__(
 
-      my_secondaries        See class docstring above. (optional)
+      my_secondaries        See class docstring above. (optional: will also be
+                            populated by communication from Secondaries)
```
I don't quite follow this comment.  Is there a method that secondaries can call to modify `self.my_secondaries`?

The comment might be clearer if you stated how `my_secondaries` may be updated/populated.

vladimir-v-diaz commented:
It's also covered in our guidelines.
https://github.com/secure-systems-lab/code-style-guidelines#dont-use-mutable-objects-as-argument-defaults

awwad commented:
Thanks (:

awwad committed:
DOC: Remove vague comment about optional arg my_secondaries

The reference implementation does not decide how this argument
is populated. It can be provided as an optional arg to the
constructor, or modified externally.

Signed-off-by: Sebastien Awwad <sebastien.awwad@gmail.com>
